src/PhpParser/Node/Value/ClassResolver.php

```diff
@@ -54,37 +54,20 @@ private function resolveFromExpression(Expression $expression): ?FullyQualified
 
     private function resolveFromClassMethod(ClassMethod $classMethod, MethodCall $methodCall): ?FullyQualified
     {
-        $class = $this->tryToResolveClassMethodStmts($classMethod);
-
-        if ($class === null) {
-            $class = $this->tryToResolveClassMethodParams($classMethod, $methodCall);
-        }
-
-        return $class;
+        return 'this' === $methodCall->var->name
+            ? $this->tryToResolveClassMethodFromThis($classMethod)
+            : $this->tryToResolveClassMethodParams($classMethod, $methodCall);
     }
 
-    private function tryToResolveClassMethodStmts(ClassMethod $classMethod): ?FullyQualified
+    private function tryToResolveClassMethodFromThis(ClassMethod $classMethod): ?FullyQualified
     {
-        // $ this -> method();
-        $stmts = $classMethod->stmts;
-        if ($stmts === null) {
-            return null;
-        }
+        $class = $classMethod->name->getAttribute(ClassLike::class)->name;
```

Comment:
```suggestion
        $class = $classMethod->name->getAttribute(AttributeKey::CLASS_NAME);
```

Reply:
This doesn't work. it is not the same thing.